src/main/scala/is/hail/annotations/Region.scala

```diff
@@ -349,6 +349,19 @@ final class Region(private var mem: Array[Byte], private var end: Long = 0) exte
           i += 1
         }
         v.leaveStruct()
+      case t: TTuple =>
```

Comment:
TTuple and TStruct have the same memory representation, and should be handled the same. Can't we write it without separate TTuple stuff in RV land?

Reply:
I couldn't figure out a nice way to combine them if we want the `PrettyVisitor` to print tuples as `(val1, val2)` instead of `{"_1" : val1, "_2" : val2}`. Ideas?

Reply:
can't we do this if the print code is handled by TStruct/TTuple? This code just visits each element in the region value.

Reply:
The visitors aren't intended to only visit fundamental types (e.g. there is visitString and visitBinary) but I didn't see a way to visit complex types without visiting its representation.

I talked this over with Jackie and we think TStruct and TTuple should both be fundamental types.  TStructBase can't be fundamental since it is abstract (can't be instantiated).  TTuple could, I guess.  Putting artificial field names on TTuples seems silly.

Reply:
if fundamental types refer to data representation, TTuple and TStruct are the _same_ fundamental type. I think maybe we shouldn't have them inherit from TStructBase, but instead have something like:

```
final class TStructBase
final class TNamedStruct {
override val fundamentalType = TStructBase(types)
}
final class TTuple {
override val fundamentalType = TStructBase(types)
}```

Reply:
I'm very uncomfortable having the most complicated RV code duplicated exactly.

Reply:
Another option is to have TStruct inherit from TTuple. We never use (or shouldn't use) field names at runtime, only field indices.

Reply:
I kind of feel like TStruct being a TTuple with some field names is the more natural direction to do things. I just wrote MatrixTable.rename and it feels kind of wrong to have to muck that deeply with the partitioner (which doesn't care so much about field names) just to rename a row key field. But having fundamental types be able to be defined as TStructs with the field names is also a nice/natural thing to do, so I don't know that that's the answer either.